src/tests/ftest/rebuild/container_create_race.py

```diff
@@ -1,284 +1,166 @@
-                self.container[0].destroy()
-
-            # Destroy the pools
-            for pool in self.pool:
-                pool.destroy(1)
-
-            self.log.info(
-                "%s: Loop %s", loop_id,
-                "passed" if errors[loop] == 0 else "failed")
-
-        self.log.info("Test %s", "passed" if sum(errors) == 0 else "failed")
+        # create pool
+        self.create_pool()
+
+        # make sure pool looks good before we start
+        info_checks = {
+            "pi_uuid": self.pool.uuid,
+            "pi_ntargets": node_qty * targets,
+            "pi_nnodes": node_qty,
+            "pi_ndisabled": 0
+        }
+        rebuild_checks = {
+            "rs_errno": 0,
+            "rs_done": 1,
+            "rs_obj_nr": 0,
+            "rs_rec_nr": 0
+        }
+        self.assertTrue(
+            self.pool.check_pool_info(**info_checks),
+            "Invalid pool information detected before rebuild")
+        self.assertTrue(
+            self.pool.check_rebuild_status(**rebuild_checks),
+            "Invalid pool rebuild info detected before rebuild")
+        self.pool.display_pool_daos_space("after creation")
+
+        # perform ior before rebuild
+        self.log.info("..(1)Start IOR before rebuild")
+        for ind in range(ior_loop):
+            self.log.info("..Starting ior run number %s", ind)
+            self.run_ior_with_pool()
+
+        # Kill the server and trigger rebuild
+        self.log.info("..(2)Starting rebuild by killing rank %s", rank)
+        self.server_managers[0].stop_ranks([rank], self.d_log, force=True)
+
+        # Wait for rebuild to start.
+        self.log.info("..(3)Wait for rebuild to start")
+        self.pool.wait_for_rebuild(True, interval=1)
+
+        # Race condition, create containers write and read during rebuild.
+        self.log.info("..(4)Create containers, write/read during rebuild")
+        self.add_containers_during_rebuild(qty=cont_qty)
+        self.access_container()
+
+        # Wait for rebuild to complete.
+        self.log.info("..(5)Wait for rebuild to finish")
+        self.pool.wait_for_rebuild(False, interval=1)
+
+        self.pool.set_query_data()
+        rebuild_status = self.pool.query_data["response"]["rebuild"]["state"]
+        self.log.info("Pool %s rebuild status:%s", self.pool.uuid,
+                      rebuild_status)
+
+        # Check for pool and rebuild info after rebuild
+        self.log.info("..(6)Check for pool and rebuild info after rebuild")
+        info_checks["pi_ndisabled"] += targets
+        rebuild_checks["rs_obj_nr"] = ">0"
+        rebuild_checks["rs_rec_nr"] = ">0"
+        self.assertTrue(
+            self.pool.check_pool_info(**info_checks),
+            "Invalid pool information detected before rebuild")
+        self.assertTrue(
+            self.pool.check_rebuild_status(**rebuild_checks),
+            "Invalid pool rebuild info detected after rebuild")
```

Comment:
How sure we are the data migrated by rebuild didn't lose data? Maybe not critical... But there is a daos container (data integrity) check which can be run to make sure rebuild did the job correctly.

Reply:
This testcase is to check the container create during restart, same as the other rebuild testcases, the pool and rebuild info counters are verified before and after the rebuild, data integrity is check with ior and io write and read.

Reply:
I understand. How do you know rebuild performed it job correctly? Doing a data integrity check with make sure the data is not corrupted after rebuild.